cameraview/src/main/java/com/otaliastudios/cameraview/CameraView.java

```diff
@@ -1555,6 +1556,38 @@ public void run() {
                 }
             });
         }
+
+        /**
+         * Log and redirect the given error information to the CameraListeners.
+         *
+         * @param exception the error cause
+         */
+        @Override
+        public void onError(final CameraException exception) {
+            // log
+            LOG.e(exception.getMessage(), exception.getCause());
+
+            // redirect
+            mUiHandler.post(new Runnable() {
+                @Override
+                public void run() {
+
+                    // all error listeners will be called, but at most one of them should actually
+                    // throw the exception
+                    int count = 0;
+                    for (CameraListener listener : mListeners) {
+                        try {
+                            listener.onError(exception);
+                        } catch (CameraException ce) {
+                            count++;
```

Comment:
can you check `if (ce == exception) then count++` ? User might do something strange during `onError` like trying to restart the camera or whatever else that might trow another exception and we don't want to catch that.

Also for the same reason please save mListeners.size() before dispatching